modules/ProvBase/Console/HardwareSupportCommand.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Modules\provbase\Console;
+
+use Carbon\Carbon;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+use Modules\ProvBase\Entities\ProvBase;
+
+class HardwareSupportCommand extends Command
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'nms:hardware-support';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'checks for hardware support of each modem and CMTS';
+
+    protected $domain_name = '';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->domain_name = ProvBase::first()->domain_name;
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command - Check for hardware support for each modem and each CMTS modules
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $this->snmp_def_mode();
+        $modems = DB::table('modem')->get();
+        $cmtses = DB::table('cmts')->get();
+        $ro_community = ProvBase::first()->ro_community;
+
+        foreach ($modems as $modem) {
+            $hostname = $modem->hostname . '.' . $this->domain_name;
+            $support_state = 'not-supported';
+            if (!isset($modem->serial_num) || $modem->serial_num === '') {
+                try {
+                    $modem->serial_num = snmpget($hostname, $ro_community, '1.3.6.1.2.1.69.1.1.4.0');
+                } catch (\Exception $exception) {
+                    $this->error($exception->getMessage());
+                }
+            }
+            $modem_serial_no_md5 = md5($modem->serial_num);
+            $contents = file_get_contents('https://support.nmsprime.com/hwsn/api.php?q=' . $modem_serial_no_md5);
+
+            if ($contents !== '') {
+                $result = json_decode($contents, true);
+                if (isset($result[$modem_serial_no_md5]) && $result[$modem_serial_no_md5] === 'valid') {
+                    $support_state = 'full-supported';
+                } elseif ((Carbon::parse($modem->created_at))->diffInWeeks(Carbon::now()) < 6) {
```

Comment:
as far as I know "created_at" is always cast to a Carbon Date by Laravel and there is also a shorthand helper now()
so this line could be simplified to
```suggestion
                } elseif (($modem->created_at->diffInWeeks(now())) < 6) {
```

Reply:
Yes you are right if we would get the records as eloquent model instances eg: Modem::all(().
In case of retrieving them as collections, the date fields are not wrapped as Carbon instance hence the "Call to a member function diffInWeeks() on string" error would occur.

Reply:
sure :man_facepalming: I forgot about that.

then no objections :grinning: